sdcm/tester.py

```diff
@@ -866,19 +867,28 @@ def set_system_auth_rf(self, db_cluster=None):
             # TODO: move this skip to siren-tools when possible
             self.log.warning("Skipping this function due this job run from Siren cloud!")
             return
-        # change RF of system_auth
-        system_auth_rf = self.params.get('system_auth_rf')
-        if system_auth_rf > 1 and not self.test_config.REUSE_CLUSTER:
-            self.log.info('change RF of system_auth to %s', system_auth_rf)
-            node = db_cluster.nodes[0]
-            credentials = db_cluster.get_db_auth()
-            username, password = credentials if credentials else (None, None)
-            with db_cluster.cql_connection_patient(node, user=username, password=password) as session:
-                session.execute("ALTER KEYSPACE system_auth WITH replication = "
-                                "{'class': 'org.apache.cassandra.locator.SimpleStrategy', "
-                                "'replication_factor': %s};" % system_auth_rf)
-            self.log.info('repair system_auth keyspace ...')
-            node.run_nodetool(sub_cmd="repair", args="-- system_auth")
+        # no need to change rf if 1 node in cluster (artifact tests)
+        if len(db_cluster.nodes) == 1:
+            self.log.debug("Skipping change rf for system_auth on single node tests")
+            return
+        if self.test_config.REUSE_CLUSTER:
+            self.log.debug("Skipping change rf of system_auth for reusing cluster")
+            return
+        node = self.db_cluster.nodes[0]
+        nodes_by_region = self.db_cluster.nodes_by_region()
+        datacenters = {}
+        for region in nodes_by_region:
+            dc_name = self.db_cluster.get_nodetool_info(nodes_by_region[region][0])['Data Center']
```

Comment:
We have `sdcm.cluster.BaseCluster.get_datacenter_name_per_region` function

Reply:
Changed to use this method. Testing fix.